salt/modules/rabbitmq.py

```diff
@@ -111,15 +117,14 @@ def _get_rabbitmq_plugin():
     This works by taking the rabbitmq-server version and looking for where it
     seems to be hidden in /usr/lib.
     '''
-    rabbitmq = __context__.get('rabbitmq-plugins')
+    global RABBITMQ_PLUGINS
 
-    if rabbitmq is None:
+    if RABBITMQ_PLUGINS is None:
         version = __salt__['pkg.version']('rabbitmq-server').split('-')[0]
-        rabbitmq = ('/usr/lib/rabbitmq/lib/rabbitmq_server-{0}'
-                    '/sbin/rabbitmq-plugins').format(version)
-        __context__['rabbitmq-plugins'] = rabbitmq
+        RABBITMQ_PLUGINS = ('/usr/lib/rabbitmq/lib/rabbitmq_server-{0}'
+                            '/sbin/rabbitmq-plugins').format(version)
```

Comment:
I'm in favour of keeping the pre-calculated path to `rabbitmqctl` in place - I believe that the logic and storage could be best served by wrapping it all in a `_get_rabbitmqctl()` function that calculates, stores and retrieves that path and calling that everywhere instead.

For example, `RABBITMQ_PLUGINS` can be set either in `__virtual__()` or in `_get_rabbitmq_plugins()`. The dual definition creates a gap where bugs will inevitably hide and there's no one source of truth that decides what that global should now be.